### PR TITLE
[substrate] Helm charts for VitalAM Frontend created along with a ReadME file for…

### PR DESCRIPTION
**Pull request by angelaalagbe**
… further instructions.

Signed-off-by: Angela Alagbe <angela.alagbe@accenture.com>

**Changelog**
- Added helm charts and a ReadME file

 

**Reviewed by**
@sownak 
 

**Linked issue**
#INFRA-21 Create helmchart to deploy VitalAM frontend

### CODE REVIEW

File: examples/substrate/vitalam-app/charts/vitalam-frontend/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```
Comment (sownak) [inline examples/substrate/vitalam-app/charts/vitalam-frontend/Chart.yaml]: Copyright is missing

File: examples/substrate/vitalam-app/charts/vitalam-frontend/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+apiVersion: app/v1
+kind: Deployment
+metadata: 
+  name: frontEnd
+  namespace: frontend-ns
```
Comment (sownak) [inline examples/substrate/vitalam-app/charts/vitalam-frontend/templates/deployment.yaml]: Why are the name and namespace hardcoded in the templates?

File: examples/substrate/vitalam-app/charts/vitalam-frontend/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      app: vitalam-fe
+  template:
+    metadata:
+      annotations:
+        checksum/service: {{ include (print $.Template.BasePath "/service.yaml") . | sha256sum }}
+      labels:
+        app: vitalam-fe
+    spec:
+      imagePullSecrets: 
+        - name: regcred
```
Comment (sownak) [inline examples/substrate/vitalam-app/charts/vitalam-frontend/templates/deployment.yaml]: no hardcoding in templates, you have to use the values.yaml file to define default values

File: examples/substrate/vitalam-app/charts/vitalam-frontend/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  name: vitalam-fe-service
+  labels:
+    vitalam-fe
+spec:
+  selector:
+    name: frontend-app
+  ports:
+    - protocol: 'TCP'
+      port: 80
```
Comment (sownak) [inline examples/substrate/vitalam-app/charts/vitalam-frontend/templates/service.yaml]: port should not be hardcoded.

File: examples/substrate/vitalam-app/charts/vitalam-frontend/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+appVersion: "2.0"
+description: A Helm chart for deploying VitalAM frontend for substrate
+name: vitalam-frontend
```
Comment (sownak) [inline examples/substrate/vitalam-app/charts/vitalam-frontend/Chart.yaml]: Also, the path is wrong, please take latest from develop branch

**Review by sownak — CHANGES_REQUESTED** [review]

File: examples/vitalam-app/charts/vitalam-frontend/values.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+##############################################################################################
+#  This Chart is a fork from https://github.com/digicatapult/vitalam-ipfs
+#  Please update if needed
+##############################################################################################
+
```
Comment (TonyRowntree) [inline examples/vitalam-app/charts/vitalam-frontend/values.yaml:6]: value file should not have hardcoded values - should be generated by the helmtemplate job using variables passed in via the ansible role

File: examples/vitalam-app/charts/vitalam-frontend/values.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+metadata:
+  name: frontEnd
+  namespace: frontend-ns
+
+config:
+  port: 80
+  logLevel: info
+  enableLivenessProbe: true
+  applicationConfig:
```
Comment (TonyRowntree) [inline examples/vitalam-app/charts/vitalam-frontend/values.yaml:26]: Add some comments on what these values are

File: examples/vitalam-app/charts/vitalam-frontend/values.yaml (hunk 6 above)
Comment (sownak) [reply examples/vitalam-app/charts/vitalam-frontend/values.yaml:6]: Also, the copyright info seems to be wrong. If you are not taking the chart from that path you specified, there is no need to add attribution

File: examples/vitalam-app/charts/vitalam-frontend/templates/service.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+##############################################################################################
+#  This Chart is a fork from https://github.com/inteli-poc/inteli-demo
+#  Please update if needed
+##############################################################################################
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: vitalam-fe-service
```
Comment (TonyRowntree) [inline examples/vitalam-app/charts/vitalam-frontend/templates/service.yaml]: can be passed in via the value or helper file

File: examples/vitalam-app/charts/vitalam-frontend/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+##############################################################################################
+#  Copyright Accenture. All Rights Reserved.
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+apiVersion: v1
+appVersion: "2.0"
+description: A Helm chart for deploying VitalAM frontend for substrate
```
Comment (kaverikhaneja) [inline examples/vitalam-app/charts/vitalam-frontend/Chart.yaml]: vitalam frontend for 'vitalam app'

File: examples/vitalam-app/charts/vitalam-frontend/templates/_helpers.tpl [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{{- define "labels.custom" }}
+  {{ range $key, $val := $.Values.metadata.labels }}
```
Comment (kaverikhaneja) [inline examples/vitalam-app/charts/vitalam-frontend/templates/_helpers.tpl:2]: Cannot find metadata.labels in values file. Can you please check once?

**Review by sownak — APPROVED** [review]
Chart looks okay

**Pull request merged**